opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -76,3 +78,100 @@ def on_end(self, span: Span) -> None:
 
     def shutdown(self) -> None:
         self.span_exporter.shutdown()
+
+
+class BatchExportSpanProcessor(SpanProcessor):
+    """Batch span processor implementation.
+
+    BatchExportSpanProcessor is an implementation of `SpanProcessor` that
+    batches ended spans and pushes them to the configured `SpanExporter`.
+    """
+
+    def __init__(
+        self,
+        span_exporter: SpanExporter,
+        max_queue_size: int = 2048,
+        schedule_delay_millis: int = 5000,
+        max_export_batch_size: int = 512,
+    ):
+        if max_queue_size <= 0:
+            raise ValueError("max_queue_size must be a positive integer.")
+
+        if schedule_delay_millis <= 0:
+            raise ValueError("schedule_delay_millis must be positive.")
+
+        if max_export_batch_size <= 0:
+            raise ValueError("max_export_batch_size must be positive.")
+
+        if max_export_batch_size > max_queue_size:
+            raise ValueError(
+                "max_export_batch_size must be less and equal to max_export_batch_size."
+            )
+
+        self.span_exporter = span_exporter
+        self.queue = queue.Queue(max_queue_size)
+        self.worker_thread = threading.Thread(target=self.worker)
```

Comment:
'daemon = True' https://docs.python.org/2/library/threading.html#thread-objects

Reply:
What is the point of using it?

Maybe allowing the process to end even if the span processor is not shut down properly?

Reply:
If you forget to shut down the SpanProcessor with damon=False, your process will not end because the span processor thread keeps idling around. With daemon=True, you get lost traces. In the spirit of https://github.com/open-telemetry/opentelemetry-specification/pull/153 I think that lost traces should be preferred to processes deadlocking on shutdown, but I have no strong opinion.